react-js/react-hooks.md

```diff
@@ -2,89 +2,284 @@
 
 ### Projected Time
 
-Example: 30-45 minutes
-- Lesson: ? min
-- Guided Practice: ?-? min
-- Independent Practice: ? min
-- Check for Understanding: ?-? min
+Example: 80-100 minutes
+- Lesson: 15-25 min
+- Guided Practice: 15-25 min
+- Independent Practice: 15-25 min
+- Check for Understanding: 15 min
 
 ### Prerequisites
-
-Here are links to lessons that should be completed before this lesson:
-
-- Example: [Git version control](../git-version-control/git-version-control.md)
+ [React JS Part 1](https://github.com/Techtonica/curriculum/blob/master/react-js/react.md)
 
 ### Motivation
 
-Here's why this topic should be learned. Strengthen this section by including a company that uses this skillset, and a screenshot example if possible.
+React is one of the most popular frameworks for front-end development. It is important to have a way to store and change component state that is efficient and easy to read. Components can have their own state, or use shared stateful logic. Before hooks, there were many different approaches to sharing stateful logic. Hooks are great because they introduce *one* common way to reuse code. They ["let you split one component into smaller functions based on what pieces are related".]([https://reactjs.org/docs/hooks-intro.html#complex-components-become-hard-to-understand](https://reactjs.org/docs/hooks-intro.html#complex-components-become-hard-to-understand))
 
 ### Objectives
 
 **Participants will be able to:**
 
-- leave this lesson knowing this
-- and how to do this on their own
-- Even more things
-- Even more things
+- Understand what pure functions and pure components are
```

Comment:
What do you feel is the relevance of pure functions and pure components are, with regards to hooks?

Reply:
in the original template the topic was 'hooks + pure functions'  @alodahl would you rather I move this section to react part 1?

Reply:
I would just revise to be functional component vs class component. We don't need to cover pure here.